openapi/publicapi_spec.json

```diff
@@ -2290,27 +2290,6 @@
           }
         }
       },
-      "PaymentLinksForSearch" : {
-        "type" : "object",
-        "description" : "links for search payment resource",
-        "properties" : {
-          "cancel" : {
-            "$ref" : "#/components/schemas/PostLink"
-          },
-          "capture" : {
-            "$ref" : "#/components/schemas/PostLink"
-          },
-          "events" : {
-            "$ref" : "#/components/schemas/Link"
-          },
-          "refunds" : {
-            "$ref" : "#/components/schemas/Link"
-          },
-          "self" : {
-            "$ref" : "#/components/schemas/Link"
```

Comment:
Having run a test it looks like more links will be included, but they will be null where applicable:
```
            "_links": {
                "self": {
                    "href": "http://publicapi.url/v1/payments/0",
                    "method": "GET"
                },
                "next_url": null,
                "next_url_post": null,
                "auth_url_post": null,
                "events": {
                    "href": "http://publicapi.url/v1/payments/0/events",
                    "method": "GET"
                },
                "refunds": {
                    "href": "http://publicapi.url/v1/payments/0/refunds",
                    "method": "GET"
                },
                "cancel": {
                    "href": "http://publicapi.url/v1/payments/0/cancel",
                    "method": "POST"
                },
                "capture": null
```
A few things to note here:
* [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS) states that: "With HATEOAS, a client interacts with a network application whose application servers provide information dynamically through hypermedia. A REST client needs little to no prior knowledge about how to interact with an application or server beyond a generic understanding of hypermedia." It follows that having a nulls for `next_url`, `next_url_post`, and `auth_url_post` are ok because the client will explicitly know that those links don't exist.
* It seems like we're already returning a [capture link, which could be null](https://github.com/alphagov/pay-publicapi/blob/master/src/main/java/uk/gov/pay/api/model/links/PaymentLinksForSearch.java#L31), so we're already returning values for links.
* I could exclude null fields by using `@JsonInclude(Include.NON_NULL)` (haven't tried at time of writing as it's near the end of the day.)

Reply:
Am not fussed though, happy to keep PaymentLinksForSearch if you feel it's better.

Reply:
`@JsonInclude(JsonInclude.Include.NON_EMPTY)` on `PaymentLinks` class will ensure empty values don't get returned.